akka-stream-tests/src/test/java/akka/stream/javadsl/FlowThrottleTest.java

```diff
@@ -0,0 +1,94 @@
+package akka.stream.javadsl;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorRef;
+import akka.japi.Pair;
+import akka.japi.function.Function;
+import akka.stream.*;
+import akka.testkit.JavaTestKit;
+import org.junit.ClassRule;
+import org.junit.Test;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import akka.testkit.AkkaSpec;
+
+public class FlowThrottleTest extends StreamTest {
+    public FlowThrottleTest() {
+        super(actorSystemResource);
+    }
+
+    @ClassRule
+    public static AkkaJUnitActorSystemResource actorSystemResource =
+            new AkkaJUnitActorSystemResource("ThrottleTest", AkkaSpec.testConf());
+
+    @Test
+    public void mustWorksForTwoStreams() throws Exception {
+        final JavaTestKit probe1 = new JavaTestKit(system);
+        final JavaTestKit probe2 = new JavaTestKit(system);
+
+        final Function<Throwable, Supervision.Directive> decider = exc -> {
+            return Supervision.restart();
+        };
+
+        final Materializer mat1 = ActorMaterializer.create(ActorMaterializerSettings.create(system).withSupervisionStrategy(decider), system);
+        final Materializer mat2 = ActorMaterializer.create(ActorMaterializerSettings.create(system).withSupervisionStrategy(decider), system);
+
+        final String name1 = "Pipe 1: ";
+        final String name2 = "Pipe 2: ";
+
+        Source<Object, NotUsed>  pipe1 = createPipeLine(name1);
```

Comment:
this and line 50 creates two different throttle instances so it does not test reusability.